### ISSUE TITLE
Several problems in RLOOTrainer

### ISSUE THREAD

**Issue by serendipity800** [huggingface/trl#2316]
### System Info

main

### Information

- [X] The official example scripts
- [ ] My own modified scripts

### Tasks

- [X] An officially supported task in the `examples` folder
- [ ] My own task or dataset (give details below)

### Reproduction

1. metrics["loss/value_avg"] = self.accelerator.gather(vf_loss_stats).mean().item()
this metrics is useless since we don't use value func in RLOO

2. self.state.epoch = self.state.episode / self.train_dataset_len  # used by self.log 
This will cause problems on calculation since the true epoch count is (self.state.episode / (args.rloo_k * self.train_dataset_len) Since every instruction is repeated args.rloo_k times

3. Multiple self.state.global_step += 1
will cause the saving process to go wrong

### Expected behavior

this to be right
